src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -503,21 +510,34 @@ public void SendAsync_ExpectedDiagnosticSynchronousExceptionActivityLogging()
                     using (HttpClientHandler handler = CreateHttpClientHandler(useSocketsHttpHandlerString))
                     using (HttpClient client = new HttpClient(handler))
                     {
+                        // Set a https proxy.
+                        handler.Proxy = new WebProxy($"https://{Guid.NewGuid()}.com", false);
+                        HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, $"http://{Guid.NewGuid()}.com");
+
                         if (bool.Parse(useSocketsHttpHandlerString))
                         {
-                            // Forces a synchronous exception for SocketsHttpHandler
-                            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, $"http://{Guid.NewGuid()}.com");
-                            request.Version = new Version(0, 0);
-
-                            Assert.ThrowsAsync<NotSupportedException>(() => client.SendAsync(request)).Wait();
+                            // Forces a synchronous exception for SocketsHttpHandler.
+                            // SocketsHttpHandler only allow http scheme for proxies.
+
+                            // We cannot use Assert.Throws<Exception>(() => { SendAsync(...); }) to verify the
+                            // synchronous exception here, because DiagnosticsHandler SendAsync method has async
+                            // modifier, and returns Task. In this case, the Task will complete synchronously.
+                            Task sendTask = client.SendAsync(request);
+                            Assert.True(sendTask.IsFaulted);
```

Comment:
I don't understand this.  If a task is returned into the 'sendTask' variable, then it is NOT testing a sync exception from the .SendAsync() method.  It is an async exception because a task is returned and the task has faulted.

Reply:
I should make the comment clearer. The details are in PR description. 

> DiagnosticsHandler SendAsync method has async modifier, and returns Task<HttpResponseMessage>. If not await that call, the current test method will continue run before the call is completed: Assert.Throws<Exception>(() => { client.SendAsync(request); }) will not observe any Exception (diagnosticListenerObserver will observe the sync Exception though).

Note the sync Exception will observed by `diagnosticListenerObserver `, but if we want to valid the Exception (and we do), we will need to wait for the Task to complete synchronously. That's why this pattern is chose here.

Reply:
>I should make the comment clearer. The details are in PR description.

Someone reading this test code 1 year from now won't know about the PR description. So, it would be best if the comment could be clearer about why/how this test actually works.